core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -5,158 +5,148 @@
 package akka.kafka.internal
 
 import java.util
+import java.util.regex.Pattern
 
 import akka.Done
-import akka.actor.{Actor, ActorRef, Cancellable, Status}
+import akka.actor.{Actor, ActorRef, Cancellable, Props, Status}
 import akka.event.LoggingReceive
-import akka.kafka.internal.ConsumerStage.{CommittableOffsetBatchImpl, Committer}
-import akka.kafka.scaladsl.Consumer.PartitionOffset
-import akka.util.Timeout
+import akka.kafka.ConsumerSettings
 import org.apache.kafka.clients.consumer._
 import org.apache.kafka.common.TopicPartition
 
-import scala.concurrent.{ExecutionContext, Future}
-import scala.concurrent.duration._
+import scala.collection.JavaConverters._
+import scala.concurrent.Promise
 
 object KafkaConsumerActor {
```

Comment:
Actor based implementation of kafka consumer. Used to provide multithread access to same consumer